### ISSUE TITLE
error from ant build in circleCI

### ISSUE THREAD

**Issue by pbergner** [CodeScience/circle-start#1]
I keep getting the following error when it tries to build, any suggestions? I believe everything mirrors the setup provided in the repo.

    Build:
     [echo] Continuous Integration Build
      [taskdef] Could not load definitions from resource net/sf/antcontrib/antlib.xml. It could not be found.
     [echo] Branch: development
     [echo] ${build.cmd}

    BUILD FAILED
    /home/ubuntu/Synapse/build.xml:12: The following error occurred while executing this line:
    Target "${build.cmd}" does not exist in the project "null".

**Comment by dshahin** [comment]
@pbergner do you have your java files and xml files setup in your repo?
Can you provide a link to your repo for reference?

**Comment by pbergner** [comment]
Its a private repo so unfortunately I can't. The java files I downloaded from your repo and put in a dir called /lib. My build.xml, build.properties, and circle.yml are also copies from your repo as well. Except I changed build-master.properties to just build.properties and the reference to it in build.xml to be hardcoded for now. <property file="build.properties" />

**Comment by dshahin** [comment]
the property files should always be named after the branch they belong to. 
so:
build.master.properties for the master branch
build.feature-branch-name.properties for a branch named "feature-branch-name"

not sure if that is at the root of your problem, but I would try it and lmk if it helps.

**Comment by pbergner** [comment]
Tried that didn't seem to make a diff. unfortunately. I tried rebuilding without cache and now I'm getting this:

Buildfile: /home/ubuntu/Synapse/build.xml

    Build:
     [echo] Continuous Integration Build
    [taskdef] Could not load definitions from resource net/sf/antcontrib/antlib.xml. It could not be found.
     [echo] Branch: development
     [echo] DeployAndCheckOnly

    DeployAndCheckOnly:
     [echo] Deploy Code CheckOnly in Org with username: ${env.MASTER_USERNAME}

    BUILD FAILED
    /home/ubuntu/Synapse/build.xml:12: The following error occurred while executing this line:
    /home/ubuntu/Synapse/build.xml:16: Problem: failed to create task or type    antlib:com.salesforce:deploy
    Cause: The name is undefined.
    Action: Check the spelling.
    Action: Check that any custom tasks/types have been declared.
    Action: Check that any <presetdef>/<macrodef> declarations have taken place.
    No types or tasks have been defined in this namespace yet

    This appears to be an antlib declaration. 
    Action: Check that the implementing library exists in one of:
        -/usr/share/ant/lib
        -/home/ubuntu/.ant/lib
        -a directory added on the command line with the -lib argument

**Comment by pbergner** [comment]
Do I need to set any env. variables besides the username and password?

**Comment by dshahin** [comment]
that should be all you need.

can you list the contents of your /lib directory?
and confirm that it is at root of the project?

**Comment by pbergner** [comment]
Ok, so I feel stupid now, when I copied the .jar files I accidentally downloaded html files by right clicking and saving instead of the actual jar files and they saved with the .jar extension. I guess I should have noticed the file size. Anyway thanks!

**Comment by dshahin** [comment]
glad you got it sorted out, @pbergner

**Comment by pbergner** [comment]
Hey @dshahin one last question, could you summarize real quick what is the diff between DeployAndCheckOnly and DeployAndRunAllTests? 

It seems like it pushed my changes to my other org from just a pull request, I don't want it to do that unless a pull request is merged. Any insight into how its setup to work would be awesome!

**Comment by dshahin** [comment]
@pbergner  DeployAndRunAllTests runs all Salesforce tests and verifies their success before pushing the code.  I highly recommend using it for any branch that is for production or QA.
